firewood/src/db.rs

```diff
@@ -462,7 +471,7 @@ impl Db {
     }
 
     /// Open a database.
-    fn new_internal<P: AsRef<Path>>(db_path: P, cfg: DbConfig) -> Result<Self, DbError> {
+    pub fn new_internal<P: AsRef<Path>>(db_path: P, cfg: DbConfig) -> Result<Self, DbError> {
```

Comment:
nit: Let's not make this available outside the crate. We have a visibility problem already.
```suggestion
    pub(crate) fn new_internal<P: AsRef<Path>>(db_path: P, cfg: DbConfig) -> Result<Self, DbError> {
```